src/lib.rs

```diff
@@ -126,6 +127,34 @@ impl DeviceManager {
                             Aggregation::Individual(AstarteType::String(command)),
                         ) => commands::execute_command(command),
 
+                        (
+                            "io.edgehog.devicemanager.config.Telemetry",
+                            ["request", interface_name, what],
+                            Aggregation::Individual(data),
+                        ) => match (*what, data) {
+                            ("enable", AstarteType::Boolean(enabled)) => {
+                                self.telemetry.set_enabled(interface_name, *enabled).await;
+                            }
+
+                            ("enable", AstarteType::Unset) => {
+                                self.telemetry.unset_enabled(interface_name).await;
+                            }
+
+                            ("periodSeconds", AstarteType::LongInteger(period)) => {
+                                self.telemetry
+                                    .set_period(interface_name, *period as u64)
+                                    .await;
+                            }
+
+                            ("PeriodSeconds", AstarteType::Unset) => {
+                                self.telemetry.unset_period(interface_name).await;
+                            }
+
+                            _ => {
+                                warn!("Received malformed data from io.edgehog.devicemanager.config.Telemetry: {clientbound:?}");
+                            }
```

Comment:
Move this part to a function of  the telemetry module,  something like you made in `ota_event` function. 
A possible name should be `telemetry_config_event`.